### PR TITLE
Datetime 

### PR DESCRIPTION
**Pull request by Kawthar1012**

### CODE REVIEW

File: JUnit_tests/DummyTest.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DummyTest {
+
+    @Test
+    public void testSomething() {
+        assertEquals(4, new Long(4));
+    }
+
+}
```
Comment (majafritsch) [inline JUnit_tests/DummyTest.java]: Perhaps remove this file since it seems redundant

File: JUnit_tests/test_DateTime.java [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+- change MAXYEAR 999 -> 9999 (did the same in Date.java)
+- change condition microseconds 100000 -> 1000000
+- possible change in week day because of tyoes cast
+- add __int__ to objects in weekday()
+
+Possible implementations/functions:
+- operations <,>,= to compare dates (check lt,le,eq,... in Int.java)
+- now(), utcnow(), fromtimestamp(timestamp)
+- time()
+ */
```
Comment (majafritsch) [inline JUnit_tests/test_DateTime.java]: Maybe remove these comments since they probably don't make sense to someone outside this course. This would be better suited in the pull request description.

File: JUnit_tests/test_DateTime.java [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+    public void testDateSuccess () {
+        org.python.types.Int[] args = dateTimeArgs(2021,9,15,0,0,0,0);
+        java.util.Map<java.lang.String, org.python.Object> kwargs;
+        kwargs = new java.util.HashMap<java.lang.String, org.python.Object>();
+        DateTime test = new DateTime(args,kwargs);
+        assertEquals("2021-09-15",test.date().__str__().toString());
+    }
+
+    @Test
+    public void testTodayFail () {
```
Comment (majafritsch) [inline JUnit_tests/test_DateTime.java]: It seems like this test is failing. Perhaps it is intended, but maybe there is a way to expect failure somehow, so that the test passes

File: JUnit_tests/test_DateTime.java [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+    @Test
+    public void testFromOrdinal () {
+        DateTime test = (DateTime) DateTime.today();
+        long ordinal = 738052;
+//        Equivalent to 2021-09-19 -> 738052
+        assertEquals("2021-09-19", test.fromOrdinal(ordinal));
+    }
+
+    @Test
+    public void testToOrdinal () {
```
Comment (majafritsch) [inline JUnit_tests/test_DateTime.java]: This test also seems to fail.

File: python/common/org/python/stdlib/datetime/DateTime.java [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+    public String toOrdinal (DateTime date) {
+        long year = ((org.python.types.Int) date.year.__int__()).value;
+        long month = ((org.python.types.Int) date.month.__int__()).value;
+        long day = ((org.python.types.Int) date.day.__int__()).value;
+        long dayInNormalYear = 365;
+
+        return String.valueOf((year-1)*dayInNormalYear + checkDaysBeforeMonth(month,year) + day + checkHowManyLeapYear(year));
+    }
+
+    public static org.python.types.Bool __lt__ (DateTime date, DateTime date2) {
```
Comment (majafritsch) [inline python/common/org/python/stdlib/datetime/DateTime.java:327]: We noticed that you declared the comparison methods as static. We did not declare ours as static, but we're not sure which way is the correct way. Maybe we could discuss this at the sprint review and our next meeting.

File: JUnit_tests/test_DateTime.java [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+        java.util.Map<java.lang.String, org.python.Object> kwargs = dateTimeKwargs(2021,9,15,10,0,01,2759);
+        DateTime test = new DateTime(args, kwargs);
+        assertEquals("2021-09-15 10:00:01.002759",test.__str__().toString());
+    }
+
+    @Test
+    public void testInstanceError1 () {
+        Assertions.assertThrows(SyntaxError.class, () -> {
+            org.python.types.Int[] args = dateTimeArgs(2021,9,15,10,00,01,2759);
+            //java.util.Map<java.lang.String, org.python.Object> kwargs = dateTimeKwargs(2021,9,16,10,00,01,2759);
```
Comment (majafritsch) [inline JUnit_tests/test_DateTime.java]: Remove comment

File: JUnit_tests/test_DateTime.java [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+        java.util.Map<java.lang.String, org.python.Object> kwargs2 = dateTimeKwargs(2021,9,15,10,0,01,2759);
+        DateTime test2 = new DateTime(args2,kwargs2);
+        assertEquals(Bool.TRUE,DateTime.__eq__(test,test2));
+    }
+
+    @Test
+    public void testFromOrdinal () {
+        DateTime test = (DateTime) DateTime.today();
+        long ordinal = 738052;
+//        Equivalent to 2021-09-19 -> 738052
```
Comment (majafritsch) [inline JUnit_tests/test_DateTime.java]: Remove comment

**Review by majafritsch — COMMENTED** [review]
Great job!! We have now reviewed your pull request and have some comments that you could look in to.

- Remove .jar and .class-files from pull request
- Maybe remove the Date.java and TimeDelta.java from the pull request if you haven't made changes to them and don't need them for your own module.
- We are a bit unsure about the _python/common/python/datetime.java_  file since it is not located in _python/common/org/python/stdlib/datetime/ _. Is it supposed to be included in the pull request?
- Review the indentations of all the functions. It seems like they are not indented with four whitespaces.
- In some function declarations, if statements and for loops there seems to be a whitespace missing next to the parentheses, e.g. on row 245 and 250 in _python/common/org/python/stdlib/datetime/DateTime.java_   (_ our convention says: _if (delta == null) {_
- Some methods have CamelCased names, but our conventions states that methods should be written in snake_case.
- We think it would be nice with a few more test cases for each of methods to ensure that they work correctly. It seems like they do (:D), but when we looked at the code we felt the urge to add more tests with some edge cases just because we weren't sure they were handled. 

Apart from these things we think you've done a great job! :)

**Comment by Kawthar1012** [comment]
Thank you for your review! We deleted everything that was unnecessary and added test as suggested. According to our IDE, the indentations have indeed four white spaces so it might be a difference depending on which one you use to see the files.

**Comment by Kawthar1012** [comment]
Got what you meant by the indentations, it is now fixed. Thank you!
